src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -578,22 +649,29 @@ module.exports = function<T, P, I, TI, C>(config : HostConfig<T, P, I, TI, C>) {
     }
   }
 
-  function scheduleWork(root : FiberRoot) {
-    if (defaultPriority === SynchronousPriority) {
-      throw new Error('Not implemented yet');
+  function scheduleWork(root : FiberRoot, priorityLevel : ?PriorityLevel) {
+    if (priorityLevel == null) {
+      priorityLevel = priorityContext;
     }
 
-    if (defaultPriority === NoWork) {
+    if (priorityLevel === NoWork) {
       return;
-    }
-    if (defaultPriority > AnimationPriority) {
-      scheduleDeferredWork(root, defaultPriority);
+    } else if (priorityLevel === SynchronousPriority) {
+      scheduleSynchronousWork(root);
+    } else if (priorityLevel <= AnimationPriority) {
+      scheduleAnimationWork(root, priorityLevel);
+    } else {
+      scheduleDeferredWork(root, priorityLevel);
       return;
     }
-    scheduleAnimationWork(root, defaultPriority);
   }
 
-  function scheduleUpdate(fiber: Fiber, priorityLevel : PriorityLevel): void {
+  function scheduleUpdate(fiber: Fiber, priorityLevel : ?PriorityLevel): void {
```

Comment:
It looks like we're not really scheduling updates with any priorityLevel so we can just remove this argument completely to simply things.